src/core/bittorrent/session.cpp

```diff
@@ -813,11 +813,12 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
 
     // Remove it from session
     if (deleteLocalFiles) {
-        QDir saveDir(torrent->actualSavePath());
-        if ((saveDir != QDir(m_defaultSavePath)) && (saveDir != QDir(m_tempPath))) {
-            m_savePathsToRemove[torrent->hash()] = saveDir.absolutePath();
-            qDebug() << "Save path to remove (async): " << saveDir.absolutePath();
-        }
+        QString tmp = torrent->filePath(0);
+        tmp.truncate(tmp.indexOf("/"));
+        QString first_dir = torrent->actualSavePath() + tmp;
+        if(first_dir != torrent->actualSavePath())
```

Comment:
I think you should avoid calling `torrent->actualSavePath()` twice. Instead do `if (!tmp.isEmpty())`